tests/test_fakes.py

```diff
@@ -1255,6 +1255,24 @@ def test_transactions(conn: snowflake.connector.SnowflakeConnection):
         assert cur.fetchall() == [("Statement executed successfully.",)]
 
 
+def test_trim_cast_varchar(conn: snowflake.connector.SnowflakeConnection):
+    with conn.cursor() as cur:
+        cur.execute("create or replace table trim_cast_varchar(id number, name varchar);")
+        cur.execute("insert into trim_cast_varchar(id, name) values (1, '  name 1  '), (2, 'name2   ');")
+        cur.execute("select trim(id), trim(name) from trim_cast_varchar;")
+
+        assert cur.fetchall() == [("1", "name 1"), ("2", "name2")]
+
+    with conn.cursor() as cur:
+        cur.execute("create or replace table trim_cast_varchar_variant_field(data variant);")
+        cur.execute(
+            """insert into trim_cast_varchar_variant_field(data) values ('{"k1": "   v11  "}'),('{"k1": 21}');"""
+        )
```

Comment:
This fails for me on a real Snowflake instance with:
> ProgrammingError: 002023 (22000): SQL compilation error:
Expression type does not match column data type, expecting VARIANT but got VARCHAR(18) for column DATA

Am guessing we need to convert it to variant first?

Reply:
My bad again, updated.

As on #78 tests seems to fail due to -guessing- sqlalchemy version
https://github.com/tekumara/fakesnow/pull/78#discussion_r1566809604